apps/language_server/test/providers/document_symbols_test.exs

```diff
@@ -30,6 +30,27 @@ defmodule ElixirLS.LanguageServer.Providers.DocumentSymbolsTest do
         after
           :ok
         end
+        defmacro customdef_remote(call, do: block) do
+          quote do
+            Kernel.def(unquote(call), do: unquote(block))
+          end
+        end
+        defmacro customdef_local(call, do: block) do
+          quote do
+            customdef_remote(unquote(call), do: unquote(block))
+          end
+        end
+        defmacro customdef_kernel(call, do: block) do
+          quote do
+            def(unquote(call), do: unquote(block))
+          end
+        end
```

Comment:
These are the base cases which I could detect easily, which are local macros that either use the Kernel.def/defp/etc format, the implicit import format for the Kernel defs, or another custom def in itself, hence the respective names _remote, _kernel and _local